### PR TITLE
try to calculate direction of BG if missing

### PR DESCRIPTION
**Pull request by PoweRGbg**

### CODE REVIEW

**Comment by MilosKozak** [comment]
It looks much better. Could you add some tests?

**Comment by PoweRGbg** [comment]
@MilosKozak Ready for merge ?

**Comment by MilosKozak** [comment]
created PR with cleanup https://github.com/PoweRGbg/AndroidAPS/pull/68 . Merge it pls
but it still need update. The call for direction calculation may be performed for older records as well so it's not simply possible use last glucose status

File: app/src/test/java/info/nightscout/androidaps/db/BgReadingTest.java [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+        assertTrue(copy.isEqual(bgReading));
+        assertFalse(copy.isEqual(new BgReading()));
+    }
+
+    @Test
+    public void calculateDirection() {
+        List<BgReading> bgReadingsList = null;
+        AAPSMocker.mockDatabaseHelper();
+
+        when(MainApp.getDbHelper().getAllBgreadingsDataFromTime(anyLong(),anyBoolean())).thenReturn(bgReadingsList);
```
Comment (MilosKozak) [inline app/src/test/java/info/nightscout/androidaps/db/BgReadingTest.java]: this `when` is necesary only once

Comment (PoweRGbg) [reply app/src/test/java/info/nightscout/androidaps/db/BgReadingTest.java]: When I remove the others - test fails

**Pull request merged**